### PR TITLE
Do not create an intermediate modified.txt file.

### PR DESCRIPTION
**Pull request by teo-tsirpanis**
The contents of `modified.txt` are kept in memory instead.
This increases performance, without affecting debuggability in any way.

### CODE REVIEW

**Comment by teo-tsirpanis** [comment]
OK, it's now ready. I made the output file prettier.

File: Parser.py [hunk 1]
```diff
@@ -119,7 +106,7 @@ def signs_missing_from_of(modified_lines):
     for variables in range(len(of_range)):
         if variables == 0:
             matched = re.match("^([-+]?)((\d+(\.\d+)?)?)(\*?)x(\d+)", temp_group)
-        if variables != 0:
+        else:
```
Comment (YorgosBas) [inline Parser.py:109]: Also in line 139 of changed (152 in original)

File: Parser.py [hunk 2]
```diff
@@ -257,20 +244,14 @@ def extract_constrain_coefficients(modified_lines, range1):
...
-        if re.findall(">=", modified_lines[m]):
-            index = modified_lines[m].find("=")
-        elif re.findall("=", modified_lines[m]):
-            index = modified_lines[m].find("=")
-        elif re.findall("<=", modified_lines[m]):
-            index = modified_lines[m].find("=")
+        index = modified_lines[m].find("=")
         right_part = modified_lines[m][index + 1:]
-        list.append(right_part)
+        list.append(int(right_part))
```
Comment (YorgosBas) [inline Parser.py:249]: ValueError: invalid literal for int() with base 10: '-8,'

works well if int() is removed and reverted to original form

Comment (teo-tsirpanis) [reply Parser.py:249]: It works on me. Where did it find the comma after the 8? Commas where removed from the modified file (see line 41 from the modified).

> works well if int() is removed and reverted to original form

If we remove `int()`, the number will be stored as a string; something we definitely not want.

File: Parser.py (hunk 1 above)
Comment (teo-tsirpanis) [reply Parser.py:109]: image

PyCharm detects code duplication. I thought the `else` would fix it, but if I change it in line 152 as well, it warns us again. The two functions have to be refactored somehow... 😕

File: Parser.py (hunk 2 above)
Comment (YorgosBas) [reply Parser.py:249]: I did made the changes but somehow it gives me a warning

But that should be right

Comment (YorgosBas) [reply Parser.py:249]: I did made the changes but somehow it gives me a warning

But that should be right

Edit: fixed

**Comment by YorgosBas** [comment]
Passed tests

**Pull request merged**